composer.json

```diff
@@ -2,11 +2,13 @@
     "name": "mailgun/mailgun-php",
     "description": "The Mailgun SDK provides methods for all API functions.",
     "require": {
-        "guzzlehttp/guzzle": "~5.0"
+        "guzzlehttp/psr7": "~1.2",
+        "happyr/http-auto-discovery": "0.1.*"
```

Comment:
What about using only ``php-http/discovery``? IMHO we should be careful adding dependencies. As noted in the readme of happyr/http-auto-discovery, this can be replaced with only a few lines of code. (I noticed you are the author of the library, so you probably wrote that yourself :-))

https://github.com/Happyr/HttpAutoDiscovery